### PR TITLE
DAOS-3339 dmg: evice pool if destory pool forcely

### PR DESCRIPTION
**Pull request by wangdi1**
Let's evict pool if the pool is being destroyed
forcely.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/utils/dmg.c [hunk 1]
```diff
@@ -259,6 +265,28 @@ destroy_hdlr(int argc, char *argv[])
...
+		if (svc == NULL) {
+			fprintf(stderr, "failed to parse service ranks\n");
+			return 2;
+		}
+
+		rc = daos_pool_evict(pool_uuid, sysname, svc, NULL /* ev */);
+		if (rc != 0) {
+			fprintf(stderr, "failed to evict pool: %d\n", rc);
+			return rc;
+		}
```
Comment (johannlombardi) [inline src/utils/dmg.c]: dmg will be eventually removed, so we need a solution for daos_shell too.
For daos_shell, could we just add a force flag on destroy and call on the server via drpc evict and then destroy? @tanabarr any thoughts?

Comment (wangdi1) [reply src/utils/dmg.c]: Indeed, probably evict from the server side would be a better choice. But I am not an expert for those dRPC stuff. Could some one from metadata team take care of this?

**Comment by liw** [comment]
I’ve taken over the JIRA ticket.

File: src/utils/dmg.c (hunk 1 above)
Comment (tanabarr) [reply src/utils/dmg.c]: will probably need to split ds_pool_evict_handler into RPC specific and internal functions then call internal from the dRPC handler. then would be executed as you mention @johannlombardi

**Pull request closed without merging**